srcts/src/components/_shinyResizeObserver.ts

```diff
@@ -0,0 +1,138 @@
+   * or the card body is expanded full screen. It performs two primary tasks:
+   *
+   * 1. Dispatches a `resize` event on the window object. This is necessary to
+   *    ensure that Shiny outputs resize appropriately. In general, the window
+   *    resizing is throttled and the output update occurs when the transition
+   *    is complete.
+   * 2. If an output with a resize method on the output binding is detected, we
+   *    directly call the `.onResize()` method of the binding. This ensures that
+   *    htmlwidgets transition smoothly. In static mode, htmlwidgets does this
+   *    already.
+   *
+   * @note
+   * This resize observer also handles race conditions in some complex
+   * fill-based layouts with multiple outputs (e.g., plotly), where shiny
+   * initializes with the correct sizing, but in-between the 1st and last
+   * renderValue(), the size of the output containers can change, meaning every
+   * output but the 1st gets initialized with the wrong size during their
+   * renderValue(). Then, after the render phase, shiny won't know trigger a
+   * resize since all the widgets will return to their original size (and thus,
+   * Shiny thinks there isn't any resizing to do). The resize observer works
+   * around this by ensuring that the output is resized whenever its container
+   * size changes.
+   * @constructor
+   */
+  constructor() {
+    this.resizeObserverEntries = [];
+    this.resizeObserver = new ResizeObserver((entries) => {
+      const resizeEvent = new Event("resize");
+      window.dispatchEvent(resizeEvent);
+
+      const resized = [] as HTMLElement[];
+
+      for (const entry of entries) {
+        if (!(entry.target instanceof HTMLElement)) continue;
+        if (!entry.target.querySelector(".shiny-bound-output")) continue;
+
+        entry.target.querySelectorAll(".shiny-bound-output").forEach((el) => {
+          if (!(el instanceof HTMLElement)) return;
+          if (resized.includes(el)) return;
+
+          const { binding, onResize } = $(el).data("shinyOutputBinding");
+          if (!binding || !binding.resize) return;
+
+          // if this output is owned by another observer, skip it
+          const owner = (el as any).shinyResizeObserver;
+          if (owner && owner !== this) return;
+
+          // trigger immediate resizing of outputs with a resize method
+          onResize(el);
+          // only once per output and resize event
+          resized.push(el);
+          // mark this output as owned by this shinyResizeObserver instance
+          (el as any).shinyResizeObserver = this;
+        });
+
+        // set plot images to 100% width/height temporarily during the transition
+        entry.target
+          .querySelectorAll('.shiny-plot-output img:not([width="100%"])')
+          .forEach((el) => {
+            if (!(el instanceof HTMLImageElement)) return;
+            el.setAttribute("width", "100%");
+            el.setAttribute("height", "100%");
```

Comment:
```suggestion
```

Reply:
Done in 48ff82f9202d398a0147bb27e162a126fec0afd6